AK/FixedPoint.h

```diff
@@ -261,42 +261,22 @@ class FixedPoint {
     template<Integral I>
     bool operator>(I other) const
     {
-        if (m_value > 0)
-            return (m_value >> precision) > other || (m_value >> precision == other && (m_value & radix_mask));
-        if (other > 0)
-            return false;
-
-        return (m_value >> precision) > other || !(m_value >> precision == other && (m_value & radix_mask));
+        return (m_value >> precision) >= other || m_value >= (other << precision);
     }
     template<Integral I>
     bool operator>=(I other) const
     {
-        if (m_value > 0)
-            return (m_value >> precision) >= other || (m_value >> precision == other && (m_value & radix_mask));
-        if (other > 0)
-            return false;
-
-        return (m_value >> precision) >= other || !(m_value >> precision == other && (m_value & radix_mask));
+        return (m_value >> precision) >= other || m_value >= (other << precision);
     }
     template<Integral I>
     bool operator<(I other) const
     {
-        if (m_value > 0)
-            return (m_value >> precision) < other || !(m_value >> precision == other && (m_value & radix_mask));
-        if (other > 0)
-            return true;
-
-        return (m_value >> precision) < other || (m_value >> precision == other && (m_value & radix_mask));
+        return (m_value >> precision) < other || m_value < (other << precision);
     }
     template<Integral I>
     bool operator<=(I other) const
     {
-        if (m_value > 0)
-            return (m_value >> precision) <= other || !(m_value >> precision == other && (m_value & radix_mask));
-        if (other > 0)
-            return true;
-
-        return (m_value >> precision) <= other || (m_value >> precision == other && (m_value & radix_mask));
+        return (m_value >> precision) <= other || m_value <= (other << precision);
```

Comment:
This would suggest that `1.5fp <= 1` which is not the case.
I can't think of a much better algorithm than `*this < other || *this == other`
at the moment.